libdocument/ev-document.c

```diff
@@ -236,14 +236,16 @@ ev_document_load (EvDocument  *document,
 {
 	EvDocumentClass *klass = EV_DOCUMENT_GET_CLASS (document);
 	gboolean retval;
+	g_autofree const char *mimetype = NULL;
```

Comment:
I think drop the `const` - typically you'll make your variable here the same as the return type of what's being called - `ev_file_get_mime_type()` returns a new string, so `ev_document_load()` 'owns' it after and must free it.

The `const` keyword would be appropriate when you're getting a `const char*` in return (meaning, you *won't* own the returned string, so you must not free or modify it). Of course, in that case, you don't need the `g_autofree` either.

You could also leave the method call where it was (a g_strcmp0 argument), since with autofree you won't need to worry about memory management, but I'm ok either way. I find inline to be less readable, personally, though only in C.

Reply:
Thanks. I somehow hadn't thought of the convention that `const` denotes memory ownership. Also changed `char` to `gchar`.
I intentionally pulled the call out of the arguments to clearly separate the allocation and pointer assignment from usage. Even with `g_autofree`, we'd have to assign the `mimetype` variable for automatic cleanup.